src/vos/vos_internal.h

```diff
@@ -496,14 +498,17 @@ int obj_tree_fini(struct vos_object *obj);
 int obj_tree_register(void);
 
 /**
- * Data structure which carries the keys, epoch ranges to the multi-nested
- * btree.
+ * Single value key
  */
-struct vos_key_bundle {
-	/** key for the current tree, could be @kb_dkey or @kb_akey */
-	daos_key_t		*kb_key;
-	/** epoch of the I/O */
-	daos_epoch_t		 kb_epoch;
+struct vos_svt_key {
+	/** Epoch of entry */
+	uint64_t	sk_epoch;
+	/** Minor epoch of entry */
+	uint16_t	sk_minor_epc;
+	/** pad to 32 bits */
+	uint16_t	sk_pad1;
+	/** pad to 64 bits */
+	uint32_t	sk_pad2;
```

Comment:
I am thinking that if we can move "ir_cs_size" and "ir_cs_type" from "vos_irec_df" to "vos_svt_key", then the space overhead caused by minor epoch will be erased.

On the other hand, if we move minor_epoch from "vos_svt_key" to "ir_cs_size", there is also no space overhead, but it will cause time overhead of additional accessing PMEM when compare the "key".

So I prefer the former one if possible.

Reply:
To do this, we would need to either change to direct key implementation or fix the longstanding issue with btree not allowing multiple keys mapping to same hash.